### PR TITLE
DAOS-3984 security: Add new permissions bits

### PR DESCRIPTION
**Pull request by kjacque**
- Add permission bits to support container ACLs.
- Update daos_ace_validate() to accept the new permissions.
- Update daos_ace_from_str() and daos_ace_to_str() to
  translate the new permissions bits.
- Add new functions daos_acl_pool_validate() and
  daos_acl_cont_validate() to validate the permissions
  acceptable for pools and containers, respectively.
- Move ACL validity unit tests into their own file since
  the ACL API test file was getting too large.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/common/acl_util.c [hunk 1]
```diff
@@ -50,6 +50,13 @@
  */
 #define PERM_READ_CH		'r'
 #define PERM_WRITE_CH		'w'
+#define PERM_CREATE_CONT_CH	'c'
+#define PERM_DEL_CONT_CH	'd'
+#define PERM_GET_PROP_CH	't'
+#define PERM_SET_PROP_CH	'T'
+#define PERM_GET_ACL_CH		'a'
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/common/acl_util.c]: Correct, I double-checked it in my editor.

[the remark this answers is not in the document]
Comment (kjacque) [reply src/common/acl_util.c]: Correct, I double-checked it in my editor.

File: src/common/acl_api.c [hunk 2]
```diff
@@ -1135,63 +1193,58 @@ daos_ace_is_valid(struct daos_ace *ace)
 					DAOS_ACL_FLAG_ACCESS_FAIL |
 					DAOS_ACL_FLAG_ACCESS_SUCCESS;
 	uint64_t	valid_perms =	DAOS_ACL_PERM_READ |
-					DAOS_ACL_PERM_WRITE;
+					DAOS_ACL_PERM_WRITE |
```
Comment (jolivier23) [inline src/common/acl_api.c]: This appears to be == cont_perms | pool_perms.   If you define macros, you can reduce some duplication, and maybe avoid future omissions.

File: src/common/acl_util.c [hunk 3]
```diff
@@ -423,6 +451,20 @@ daos_ace_to_str(struct daos_ace *ace, char *buf, size_t buf_len)
...
+	if (perms & DAOS_ACL_PERM_GET_PROP)
+		rc = write_char(&pen, PERM_GET_PROP_CH, &remaining_len);
+	if (perms & DAOS_ACL_PERM_SET_PROP)
+		rc = write_char(&pen, PERM_SET_PROP_CH, &remaining_len);
+	if (perms & DAOS_ACL_PERM_GET_ACL)
+		rc = write_char(&pen, PERM_GET_ACL_CH, &remaining_len);
+	if (perms & DAOS_ACL_PERM_SET_ACL)
+		rc = write_char(&pen, PERM_SET_ACL_CH, &remaining_len);
+	if (perms & DAOS_ACL_PERM_SET_OWNER)
+		rc = write_char(&pen, PERM_SET_OWNER_CH, &remaining_len);
```
Comment (jolivier23) [inline src/common/acl_util.c]: these are the types of things I like to use FOREACH macros for.

File: src/common/acl_api.c (hunk 2 above)
Comment (kjacque) [reply src/common/acl_api.c]: Good idea, I'll address this in a follow-on PR.

File: src/common/acl_util.c (hunk 3 above)
Comment (kjacque) [reply src/common/acl_util.c]: I added a macro to simplify this in my next PR. FOREACH seemed like overkill in this case, but I did add a macro to limit the repetition.

**Pull request merged**